include/audit_events.hpp

```diff
@@ -242,7 +239,17 @@ inline void auditEvent(const crow::Request& req, const std::string& userName,
     }
 
     // encode user account name to ensure it is in an appropriate format
-    user = audit_encode_nv_string("acct", userName.c_str(), 0);
+    size_t userLen = 0;
+    char* user = audit_encode_nv_string("acct", userName.c_str(), 0);
+
+    // setup a unique_ptr to handle freeing memory from user
+    std::unique_ptr<char, void (*)(char*)> userUP(user, [](char* ptr) {
+        if (ptr != nullptr)
+        {
+            ::free(ptr);
+        }
+    });
+
     if (user == NULL)
     {
         BMCWEB_LOG_ERROR << "Error appending user to audit msg : " << errno;
```

Comment:
This `errno` would need to be used right after `audit_encode_nv_string()`

So, wouldn't this code may be

```
    size_t userLen = 0;
    char* user = audit_encode_nv_string("acct", userName.c_str(), 0);
    if (user == NULL)
    {
        BMCWEB_LOG_ERROR << "Error appending user to audit msg : " << errno;
    }
   else
   {

        // setup a unique_ptr to handle freeing memory from user
        std::unique_ptr<char, void (*)(char*)> userUP(user, [](char* ptr) {
                      if (ptr != nullptr)
                      {
                                  ::free(ptr);
                      }
        });

        userLen = std::strlen(user);

        if (userLen > bufLeft)
        {
            // Username won't fit into fixed sized buffer. Leave it off.
            BMCWEB_LOG_WARNING << "Audit buffer too small for username:"
                               << " bufLeft=" << bufLeft
                               << " userLen=" << userLen;
        }
        else
        {
            strncat(cnfgBuff, user, userLen);
            bufLeft -= userLen;
        }
    }

```

Reply:
Good catch, I should have caught that myself!